java/ingest/ingest-core/src/main/java/sleeper/ingest/job/commit/IngestJobCommitRequest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.ingest.job.commit;
+
+import sleeper.core.record.process.RecordsProcessed;
+import sleeper.core.record.process.RecordsProcessedSummary;
+import sleeper.core.statestore.FileReference;
+import sleeper.ingest.job.IngestJob;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A request to commit the results of an ingest job to the state store and job status store.
+ */
+public class IngestJobCommitRequest {
```

Comment:
This needs redesigning because we won't commit the results of a whole ingest job at once. We're just committing some files that have been output by that job, and the job may not have finished yet.